### PR TITLE
 add function delete in LevelEditorController

### PR DESCRIPTION
**Pull request by So-Gau**

### CODE REVIEW

File: src/Controller/LevelEditorController.php [hunk 1]
```diff
@@ -53,4 +53,14 @@ private function parsePost(array &$level, array &$cells, array &$errors): void
             }
         }
     }
+    public function delete(int $id)
+    {
+        $levelManager = new LevelManager();
+        $level = $levelManager->selectOneById($id);
+        if ($_SERVER['REQUEST_METHOD'] === 'POST') {
+            $levelManager = new LevelManager();
+            $levelManager->delete($level);
```
Comment (rocambille) [inline src/Controller/LevelEditorController.php:62]: strange to have these delete lines in a method returning an edit template. moreover, your delete method asks for an id as parameter, not a full Level. I guess what you want is more something like this:

    public function edit(int $id)
    {
        $levelManager = new LevelManager();
        $levelManager->selectOneById($id);
        // handle level edition, if it's a "submit" request
        return $this->twig->render('Editor/edit.html.twig');
    }

    public function delete(int $id)
    {
        $levelManager = new LevelManager();
        $levelManager->delete($id);
        // maybe redirect to an other page
    }

**Review by rocambille — CHANGES_REQUESTED** [review]

**Comment by MarvinCrepin** [comment]
done

File: src/Controller/LevelEditorController.php [hunk 2]
```diff
@@ -62,4 +62,20 @@ public function createLevel(): void
...
     }
+
+    public function delete(): void
+    {
+        $levelManager = new LevelManager();
+        $levels = $levelManager->selectAll('id');
+        $levelsId = array_column($levels, 'id');
+        if ($_SERVER['REQUEST_METHOD'] === 'POST') {
+            foreach ($_POST as $id => $value) {
+                $id = filter_var($id, FILTER_VALIDATE_INT);
```
Comment (rocambille) [inline src/Controller/LevelEditorController.php:73]: useless: the id comes from the database ;)

**Review by rocambille — CHANGES_REQUESTED** [review]

File: src/Controller/LevelEditorController.php (hunk 2 above)
Comment (MarvinCrepin) [reply src/Controller/LevelEditorController.php:73]: id comes from post not from the database

File: src/Controller/LevelEditorController.php [hunk 3]
```diff
@@ -62,4 +62,20 @@ public function createLevel(): void
...
+                if ($id !== false && in_array($id, $levelsId) && $value === 'delete') {
```
[the remark this answers is not in the document]
Comment (MarvinCrepin) [reply src/Controller/LevelEditorController.php:74]: id can be false because it comes from the form so we need to check if the input is a valid int

**Pull request merged**